src/color_picker.coffee

```diff
@@ -210,16 +210,13 @@ Ember.Widgets.ColorPicker = Ember.Component.extend
       ])
     ])
 
-  setupCustomColor: Ember.on 'init', ->
-    selectedColor = colorToHex(@get('selectedColor'))
-    return if @get('colorRows').find (row) ->
+  setCustomColor: Ember.observer(->
+    selectedColor = @get 'selectedColor'
+    selectedColor = colorToHex(selectedColor)
+    return @set('customColor', '') if @get('colorRows').find (row) ->
       selectedColor in row.invoke 'toLowerCase'
-    @set 'customColor', @get('selectedColor')
-
-  setCustomColor: Ember.observer ->
-    if @get('isCustomColorValid') is true
-      @set 'selectedColor', expandHexColor(@get('customColor'))
-  , 'customColor', 'isCustomColorValid'
+    @set 'customColor', selectedColor
+  , 'selectedColor', 'colorRows').on('init')
```

Comment:
I think it will because it's the `on` function of `Ember.observer` and not the `on` function Ember loads into the `function` prototype.